### DISCUSSION TITLE
About the v0.03

### DISCUSSION THREAD

**Discussion by VELD-Dev** [VELD-Dev/Lunacy#18] [Announcements]
The development of the v0.03 is taking lots more time than expected, as I'm rewriting and greatly optimizing the files-reading library, in order to make loading faster and more memory-efficient; On pre-ACIT games (R:FOM, RaC:ToD, RaC:QfB), the loading was taking ages and even aborted on some levels for various reasons.  

I'm rewriting the reading process in order to read the levels step-by-step instead of all mixed-up, I think that it would give a better idea of what's loading, and theoretically the changes should greatly improve loading times.  
  
It is important to note that the update may release in a few weeks from the moment I'm writing this, and could release with bugs related to these changes.  
  
The bugs that will occur for sure are, first, related to transforms: I broke them when rewriting the transforms system for a better editing support, I'll fix it in a further hotfix or maybe even in a future patch (maybe 0.04 who knows); Even though the transforms are broken, it should not be visible at first glance, but if you pay attention to detail, you may easily spot some Ties or even Mobys with broken transforms. Mentioned in #16  
  
For the v0.04, major changes will occur in the core of ReLunacy. I have detailed all those changes in #17 
  
Thanks for supporting this project, it means a lot to me !

**Comment by Mrroboto9819** [comment]
Sounds interesting, thank you for your hard work!!

**Reply by VELD-Dev** [reply]
You're welcome, and I thank you for your support 😁

**Reply by Mrroboto9819** [reply]
Just one commentary, i think the tool does not work for RFOM it only works for Resistance 2 and Resistance 3, cause RFOM have a diferent file structure or if you have test it, how can i load the files with this different file structure?

**Reply by VELD-Dev** [reply]
Good to know ! Thanks for the report. I'll inspect after I rewrote entirely the reading system

**Comment by VELD-Dev** [comment]
Devlog, the development is taking even more time than expected.

I decided not only to remake the library for this new version, but also to remake the rendering engine, and subsequently all the 3d related tools.
v0.03 will have translation, rotation and scale tools, and will have visual transparency.
I'm working hard to provide a functional version, but it's really hard, even if I have a model.

Note that v0.03 will also include **UFrags** on pre-ACIT games, like ToD and QFB !
I'm also trying to reverse volumes on pre ACIT games but I can't promise anything for any near future version.

Thanks to everyone who supports the project!
